src/link_manager/LinkManagerStateMachineActiveActive.cpp

```diff
@@ -1307,6 +1308,19 @@ void ActiveActiveStateMachine::handleDefaultRouteStateNotification(const Default
         LOGWARNING_MUX_STATE_TRANSITION(mMuxPortConfig.getPortName(), mCompositeState, nextState);
         mCompositeState = nextState;
     }
+    if (mComponentInitState.all() &&
+        oldRouteState == DefaultRoute::NA && mDefaultRouteState == DefaultRoute::OK &&
+        ps(mCompositeState) == link_prober::LinkProberState::Label::Active &&
+        mMuxPortConfig.getMode() == common::MuxPortConfig::Mode::Auto) {
+        // Let's reset the link prober state if the default route flaps faster than
+        // the link prober detection.
+        MUXLOGWARNING(
+            boost::format("%s: reset link prober state, current mux state '%s' ") %
+            mMuxPortConfig.getPortName() %
+            mMuxStateName[ms(mCompositeState)]
+        );
+        initLinkProberState(mCompositeState);
```

Comment:
This change will fix the sequence you target to fix. Is there a downside for this change? What if the link prober is already having a notification to linkmgrd outstanding? In such case, would reset cause an extra flap?

Reply:
if link prober has notified the state machine, its link prober state should be changed instead of `Active`.